WordPress/Classes/Services/SiteSegmentsService.swift

```diff
@@ -26,12 +26,16 @@ final class SiteCreationSegmentsService: LocalCoreDataService, SiteSegmentsServi
     override init(managedObjectContext context: NSManagedObjectContext) {
         self.accountService = AccountService(managedObjectContext: context)
 
+        let userAgent = WPUserAgent.wordPress()
+        let localeKey = WordPressComRestApi.LocaleKeyV2
+
         let api: WordPressComRestApi
-        if let wpcomApi = accountService.defaultWordPressComAccount()?.wordPressComRestApi {
-            api = wpcomApi
+        if let account = accountService.defaultWordPressComAccount(), let token = account.authToken {
```

Comment:
How about creating a method on the Account object called `wordPressComRestV2APi` and that will return the API object set to the correct v2 locale?

Reply:
I like this suggestion – it would remove a bit of duplication 👍

Reply:
Thanks for this suggestion! I considered it during the initial implementation; it didn't seem to help as much as I had hoped.

Because `AccountService.defaultWordPressComAccount()` is optional, you still have some duplication in the event that returns `nil`. For example:
```
let api: WordPressComRestApi
if let account = accountService.defaultWordPressComAccount() {
    api = account.wordPressComRestV2Api
} else {
    api = WordPressComRestApi.anonymousApi(userAgent: WPUserAgent.wordPress(), localeKey: WordPressComRestApi.LocaleKeyV2)
}
```

In any case, I'll proceed to make this change.